### PR TITLE
cmd: add --outs option for dag

### PR DESCRIPTION
**Pull request by JIoJIaJIu**
per https://github.com/iterative/dvc/pull/4739

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/command-reference/dag.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ restore one or more pipelines later (see `dvc repro`).
   [DOT](<https://en.wikipedia.org/wiki/DOT_(graph_description_language)>)
   format. It can be passed to third party visualization utilities.
 
+- `-o`, `--outs` - show DAG of stage outputs instead of stages.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/dag.md:47]:
```suggestion
- `-o`, `--outs` - show the graph of stage outputs instead of stage names.
```

Comment (jorgeorpinel) [reply content/docs/command-reference/dag.md:47]: But I'm not 100% sure of this. Can `--outs` actually change the graph itself? Like, showing more nodes if a stage has multiple outputs?

**Review by jorgeorpinel — COMMENTED** [review]
Sorry for the delay here...

> Can --outs actually change the graph itself? Like, showing more nodes

 If so, we probably need an example that shows this.

**Comment by shcheklein** [comment]
@JIoJIaJIu PTAL

File: content/docs/command-reference/dag.md (hunk 1 above)
Comment (JIoJIaJIu) [reply content/docs/command-reference/dag.md:47]: Yes, the graph changes completely
Without the options graph contains stages like
```
    +-------------------+  
    | data/data.xml.dvc |  
    +-------------------+  
              *            
              *            
              *            
         +---------+       
         | prepare |       
         +---------+       
              *            
              *            
              *            
        +-----------+      
        | featurize |      
        +-----------+      
         **        **      
       **            *     
      *               **   
+-------+               *  
| train |             **   
+-------+            *     
         **        **      
           **    **        
             *  *          
        +----------+       
        | evaluate |       
        +----------+       
```
but when `--outs` is added nodes are outputs (data, metrics, params)
```
                  +---------------+                    
                  | data/data.xml |                    
                  +---------------+                    
                           *                           
                           *                           
                           *                           
                  +---------------+                    
                  | data/prepared |                    
                  +---------------+                    
                           *                           
                           *                           
                           *                           
                  +---------------+                    
                  | data/features |                    
                **+---------------+****                
            ****           *           *****           
       *****               *                ****       
   ****                    *                    ****   
***                 +-----------+                   ***
  **                | model.pkl |                  **  
    **              +-----------+                **    
      **            **           **            **      
        **        **               **        **        
          **    **                   **    **          
      +-------------+             +----------+         
      | scores.json |             | prc.json |         
      +-------------+             +----------+         
```

Comment (jorgeorpinel) [reply content/docs/command-reference/dag.md:47]: OK, thanks for confirming. So maybe something like
```suggestion
- `-o`, `--outs` - show a DAG of chained dependencies and outputs instead
  of the stages themselves. The graph may be significantly different.
```
> Prob. needs formatting

Comment (jorgeorpinel) [reply content/docs/command-reference/dag.md:47]: BTW the DAG starts with a dependency (not necessarily an output) so can't the option name `outs` be misleading?

Comment (jorgeorpinel) [reply content/docs/command-reference/dag.md:47]: p.s. that's a great example too! Mind adding it as an actual Example in the doc, @JIoJIaJIu ?

**Comment by jorgeorpinel** [comment]
Ping @JIoJIaJIu 🙂

**Review by jorgeorpinel — APPROVED** [review]
LGTM and add an example separately. @JIoJIaJIu would you be able to provide the full set of commands for something like what you posted in https://github.com/iterative/dvc.org/pull/1918#discussion_r523795478 ?

**Comment by jorgeorpinel** [comment]
Merged the restyled PR.

**Pull request merged**